File: moonpoison/binary_search/374.cpp
#include<bits/stdc++.h>
using namespace std;
int guess(int num){}
class Solution {
public:
    int guessNumber(int n) 
    {
        int l=1, r=n;
        int m;
        while(l<=r)
        {
            m=l+(r-l)/2;
            if(guess(m)==-1) r=m-1;
            else if(guess(m)==1) l=m+1;
            else if(guess(m)==0) break;
        }
        return m;
    }
};
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

File: ice0208/binary-search/374.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
def guess(num: int) -> int:
    pass


class Solution:
    def bs(self, l: int, r: int) -> int:
        mid = (r - l) // 2 + l
        print(mid)
        result = guess(mid)

        if result == 0:
            return mid
        elif result == -1:
            return self.bs(l, mid - 1)
        else:
            return self.bs(mid + 1, r)

    def guessNumber(self, n: int) -> int:
        answer = self.bs(1, n)
        return answer
